src/cpuid.cpp

```diff
@@ -56,7 +56,15 @@ POSSIBILITY OF SUCH DAMAGE.
 
 
 #if TORRENT_HAS_ARM && TORRENT_HAS_AUXV
-#include <sys/auxv.h>
+#include <dlfcn.h>
+static unsigned long int _getauxval(unsigned long int type)
+{
+    using getauxval_t = unsigned long int(unsigned long int);
+    getauxval_t* getauxval = (getauxval_t*) dlsym(RTLD_DEFAULT, "getauxval");
```

Comment:
please use ``static_cast<>`` instead of a C-style cast

Reply:
do you mean ``reintrepret_cast<>`` ?

Reply:
``dlsym()`` returns a ``void*``, I'm pretty sure ``static_cast<>`` suffice.

Reply:
With the following code:
```cpp
getauxval_t p_getauxval = static_cast<getauxval_t> (dlsym(RTLD_DEFAULT, "getauxval"));
```

I stumble across the following error 
```
cpuid.cpp:64:31: error: static_cast from 'void *' to 'getauxval_t' (aka 'unsigned long (*)(unsigned long)') is not allowed
```

According to https://linux.die.net/man/3/dlsym, conversion from ``void*`` to pointer-to-function is an undefined behaviour. In fact, ``void*`` wasn't even meant to store pointer-to-function in ANSI C, but is instead a pointer-to-object storage. It seems to be the same in the C++ standard, but we can stick to POSIX for Android and assume that reinterpret_cast-ing it in a pointer-to-function is ok.

Reply:
ah. In that case, yes, I meant ``reinterpret_cast<>`` :)